src/helpers/checkpointing.py

```diff
@@ -0,0 +1,145 @@
+                    return glob.glob(checkpoint['path'] + '/generator-*.pkl')
+
+    def get_local_discriminators_paths(self, source):
+        assert source in self.get_population_sources()
+        for checkpoint in self.checkpoints_storage:
+            for local_source in checkpoint['local_discriminators']:
+                if source == local_source:
+                    return glob.glob(checkpoint['path'] + '/discriminator-*.pkl')
+
+    def get_iterations(self, source):
+        assert source in self.get_population_sources()
+        for checkpoint in self.checkpoints_storage:
+            for local_source in checkpoint['local_discriminators']:
+                if source == local_source:
+                    return checkpoint['iteration']
+
+    def get_id(self, source):
+        assert source in self.get_population_sources()
+        for checkpoint in self.checkpoints_storage:
+            for local_source in checkpoint['local_discriminators']:
+                if source == local_source:
+                    return checkpoint['id']
+
+    def get_iterations_id(self, id):
+        assert 0 <= id <= len(self.checkpoints_storage)
+        return [checkpoint['iteration'] for checkpoint in self.checkpoints_storage if checkpoint['id'] == id][0]
+
+    def get_local_generators_paths_id(self, id):
+        assert 0 <= id <= len(self.checkpoints_storage)
+        return [glob.glob(checkpoint['path'] + '/generator-*.pkl') for checkpoint in self.checkpoints_storage if
+                checkpoint['id'] == id][0]
+
+    def get_local_discriminators_paths_id(self, id):
+        assert 0 <= id <= len(self.checkpoints_storage)
+        return [glob.glob(checkpoint['path'] + '/discriminator-*.pkl') for checkpoint in self.checkpoints_storage if
+                checkpoint['id'] == id][0]
+
+    def get_discriminators_learning_rate_id(self, id):
+        assert 0 <= id <= len(self.checkpoints_storage)
+        return [checkpoint['discriminators_learning_rate'] for checkpoint in self.checkpoints_storage if checkpoint['id'] == id][0]
+
+    def get_generators_learning_rate_id(self, id):
+        assert 0 <= id <= len(self.checkpoints_storage)
+        return [checkpoint['generators_learning_rate'] for checkpoint in self.checkpoints_storage if checkpoint['id'] == id][0]
+
+    def get_adjacent_cells_id(self, id):
+        assert 0 <= id <= len(self.checkpoints_storage)
+        return \
+        [checkpoint['adjacent_individuals'] for checkpoint in self.checkpoints_storage if checkpoint['id'] == id][0]
+
+    def get_topology_details_id(self, id):
+        assert 0 <= id <= len(self.checkpoints_storage)
+        return \
+            [{'grid_size': checkpoint['grid_size'], 'position': checkpoint['position'], 'cell_info': create_cell_info(checkpoint['local_generators'][0])}  for checkpoint in self.checkpoints_storage if checkpoint['id'] == id][0]
+
+    def get_checkpoint_data_id(self, id):
+        assert 0 <= id <= len(self.checkpoints_storage)
```

Comment:
`id` is a reserved keyword in python, so I usually rename it to `_id` to not overwrite it